### PR TITLE
Split VFP classes in ebos and legacy

### PR DESCRIPTION
**Pull request by akva2**
This time with less template parameters. Turns out silly me assumed something called WellInterface.hpp and BlackoilModel.hpp went into both simulators, which is not the case, so the problem solved itself.

### CODE REVIEW

**Comment by akva2** [comment]
jenkins build this please

**Comment by GitPaean** [comment]
> Turns out silly me assumed something called WellInterface.hpp and BlackoilModel.hpp went into both simulators, 

Sorry for the messy names related to `Well` and `Wells` and `WellModel` and `WellInterface`. Big part due to we have the `flow` and `legacy` code in the same folder. Hopefully, it is much more clear when the legacy code is moved somewhere else.

File: opm/autodiff/VFPHelpersLegacy.hpp [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+
+#ifndef OPM_AUTODIFF_VFPHELPERSLEGACY_HPP_
+#define OPM_AUTODIFF_VFPHELPERSLEGACY_HPP_
+
+
+#include <opm/parser/eclipse/EclipseState/Schedule/VFPProdTable.hpp>
+#include <opm/parser/eclipse/EclipseState/Schedule/VFPInjTable.hpp>
+#include <opm/autodiff/AutoDiffHelpers.hpp>
+#include <opm/material/densead/Math.hpp>
+#include <opm/material/densead/Evaluation.hpp>
```
Comment (GitPaean) [inline opm/autodiff/VFPHelpersLegacy.hpp]: I do not think the legacy code needs the things from `densead`

Comment (akva2) [reply opm/autodiff/VFPHelpersLegacy.hpp]: indeed, copy-paste issue. will remedy after jenkins has run.

Comment (akva2) [reply opm/autodiff/VFPHelpersLegacy.hpp]: remedied now. cancelled the old build and retriggered, since there was a wait anyways..

**Comment by akva2** [comment]
jenkins build this please

**Comment by atgeirr** [comment]
All green, merging!

**Pull request merged**